datafusion/common/src/functional_dependencies.rs

```diff
@@ -97,8 +108,39 @@ impl Constraints {
                         Constraint::Unique(indices)
                     })
                 }
-                TableConstraint::ForeignKey { .. } => {
-                    _plan_err!("Foreign key constraints are not currently supported")
+                TableConstraint::ForeignKey {
+                    foreign_table,
+                    columns,
+                    name: _,
+                    on_delete: _,
+                    on_update: _,
+                    referred_columns,
+                } => {
+                    let get_indices = |c: &[Ident]| {
```

Comment:
Isn't this code looking up the 'reference columns' in the current table's schema (`df_schema`)?  I think the column references should be to the schema of `foreign_table`

Reply:
You were of course right.
I have added a test in sql_integration.rs now.
Since we don't have the schema of the referenced table, I had to change the referred_indices to be a referred_columns instead.